### PR TITLE
Adding noexcpt to isCancelled

### PR DESCRIPTION
**Pull request by vhvb1989**
fix: https://github.com/Azure/azure-sdk-for-cpp/issues/2039

### CODE REVIEW

**Review by RickWinter — APPROVED** [review]

File: sdk/core/azure-core/CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@
 ### Breaking Changes
 
 - Removed `Context::GetApplicationContext()` in favor of a new static data member `Context::ApplicationContext`.
+- Adding `noexcept` to `Context::IsCancelled()`.
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CHANGELOG.md:12]: Adding this is not a breaking change. Remove from CL.

**Review by ahsonkhan — APPROVED** [review]
Other than reverting CL, looks good.

**Pull request merged**